examples/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,124 @@
+module AffinePlaneCurveModule
+
+using Oscar, Markdown
+
+export AffinePlaneCurve, AffinePlaneCurve_degree, AffinePlaneCurve_Jacobian, AffinePlaneCurve_tangent, AffinePlaneCurve_issmoothpoint, AffinePlaneCurve_union
+
+####################################################
+#
+#We follow the lecture notes by Andreas Gathmann: https://www.mathematik.uni-kl.de/~gathmann/en/curves.php
+#
+####################################################
+mutable struct AffinePlaneCurve{S}
+  eq::Oscar.MPolyElem                      #Equation of the curve (polynomial in two variables)
+  degree::Union{String, Int64}             #degree of the equation of the curve
+  dimension::Int64                         #dimension of the curve (as a variety)
+  function AffinePlaneCurve(eq::Oscar.MPolyElem{T}) where {T <: FieldElem}
+    r = new{T}()
+    r.eq = eq
+    if nvars(parent(r.eq))!=2              #If the ambiant ring does not have two variables, show an error.
+       error("The defining equation must belong to a ring with two variables")
+    else
+       r.degree = "not computed yet"
+       r.dimension = 1                    #since C is a plane curve, the dimension is always 1
+       return r
+    end
+  end
+end
+
+function Base.show(io::IO, C::AffinePlaneCurve)
+  println(io, "Affine plane curve defined by ", C.eq)
+end
+
+######################################################
+#Compute the degree of the equation of the curve if not already known, and show it.
+
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_degree(C::AffinePlaneCurve)
+   > Given an affine plane curve C, returns the degree of the defining polynomial.
+"""
+function AffinePlaneCurve_degree(C::AffinePlaneCurve)
+  if typeof(C.degree)==Int64
+     println(C.degree)
+  else
+  C.degree=total_degree(C.eq)
+  println(C.degree)
+  end
+end
+
+######################################################
+#Compute the Jacobian Ideal of C
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_Jacobian(C::AffinePlaneCurve)
+   > Given an affine plane curve C, returns the Jacobian ideal of the defining polynomial.
```

Comment:
I believe in other places we do not have spaces before the ">" character. For consistency it would be good to follow the existing convention anyway.